src/openloco/windows/construction/OverheadTab.cpp

```diff
@@ -0,0 +1,387 @@
+                    self->widgets[widx::image].tooltip = string_ids::click_track_to_upgrade;
+            }
+        }
+
+        static string_id modString[] = {
+            string_ids::single_section,
+            string_ids::block_section,
+            string_ids::all_connected_track,
+        };
+
+        self->widgets[widx::track].text = modString[_lastSelectedTrackModSection];
+
+        common::repositionTabs(self);
+    }
+
+    // 0x0049EA3E
+    static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+    {
+        self->draw(dpi);
+        common::drawTabs(self, dpi);
+        if (_lastSelectedMods & 0xF)
+        {
+            gfx::drawpixelinfo_t* clipped = nullptr;
+            auto xPos = self->x + self->widgets[widx::image].left + 1;
+            auto yPos = self->y + self->widgets[widx::image].top + 1;
+            auto width = self->widgets[widx::image].width();
+            auto height = self->widgets[widx::image].height();
+
+            if (gfx::clip_drawpixelinfo(&clipped, dpi, xPos, yPos, width, height))
+            {
+                auto x = 0x2010;
+                auto y = 0x2010;
+
+                switch (gCurrentRotation)
+                {
+                    case 0:
+                    {
+                        auto bx = x;
+                        x = -x + y;
+                        y += bx;
+                        y >>= 1;
+                        y -= 460;
+                        break;
+                    }
+                    case 1:
+                    {
+                        x = -x;
+                        auto bx = x;
+                        x -= y;
+                        y += bx;
+                        y >>= 1;
+                        y -= 460;
+                        break;
+                    }
+                    case 2:
+                    {
+                        auto bx = x;
+                        x -= y;
+                        y = -y;
+                        y -= bx;
+                        y >>= 1;
+                        y -= 460;
+                        break;
+                    }
+                    case 3:
+                    {
+                        auto bx = x;
+                        x += y;
+                        y = -y;
+                        y += bx;
+                        y >>= 1;
+                        y -= 460;
+                        break;
+                    }
+                }
+                x -= (self->widgets[widx::image].width() / 2);
+                y -= ((self->widgets[widx::image].width() / 2) + 16);
+                clipped->x += x;
+                clipped->y += y;
+
+                _dword_E0C3E0 = (uint32_t)clipped;
+
+                x = 0x2000;
+                y = 0x2000;
+
+                auto company = companymgr::get(_playerCompany);
+                auto companyColour = company->mainColours.primary;
+                auto edi = _lastSelectedMods << 16 | 0x1D0;
```

Comment:
See comments on the functions. Clean edi and edx up into their individual components.